src/Neo/Network/P2P/Payloads/Conditions/AndCondition.cs

```diff
@@ -34,6 +35,29 @@ public class AndCondition : WitnessCondition
         public override int Size => base.Size + Expressions.GetVarSize();
         public override WitnessConditionType Type => WitnessConditionType.And;
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool Equals(AndCondition other)
+        {
+            if (ReferenceEquals(this, other))
+                return true;
+            if (other is null) return false;
```

Comment:
null check before reference equal?

Reply:
no, because of
```csharp
AndCondition a1 = null;

ReferenceEquals(a1, a1) = true;
```

Reply:
I don't understand your example, I think @Jim8y is right, if we will check null reference, it's better to do it in the first line

Reply:
So it should be this way? there are implementations in other classes that have the different order, should them be updated as well?